src/app/ReadHandler.cpp

```diff
@@ -42,7 +42,7 @@ ReadHandler::ReadHandler(Callback & apCallback, Messaging::ExchangeContext * apE
     mpExchangeMgr           = apExchangeContext->GetExchangeMgr();
     mpExchangeCtx           = apExchangeContext;
     mInteractionType        = aInteractionType;
-    mInitiatorNodeId        = apExchangeContext->GetSessionHandle()->AsSecureSession()->GetPeerNodeId();
+    mInitiatorOperationalId = apExchangeContext->GetSessionHandle()->GetPeerOperationalId();
```

Comment:
So the thing is... This is just sweeping under the rug what's really going on here.  And what's really going on here is that the callers should be ensuring group sessions never get here, so we should always have a SecureSession at this point, right?

Reply:
New code here seems different than original:
  - original was just fetching the nodeid (I see no fabric index considered)
  - new code fetches both nodeid *and* fabricid
  
For incremental changes, I would prefer to start to use `GetPeerOperationalId()->GetNodeId()` so that the change is just a rename, then follow up with logic if it makes sense.

In the usage below, it is unclear if GetAccessingFabricIndex() is identical to GetPeerOperationalId().GetFabricIndex(). I would like to be able to consider that in a _separate_ PR only. Please do usage of OperationalId (or ScopedNodeId as suggested) as a noop change  to start with.